metrics-graphite/src/main/java/com/codahale/metrics/graphite/Graphite.java

```diff
@@ -181,6 +181,11 @@ public void close() throws IOException {
         } finally {
             this.socket = null;
         }
+        
+        // if hostname is used, re-resolve DNS in connect() by clearing the address
+        if (this.hostname != null) {
```

Comment:
@arteam It looks like connect() will re-use the old address if it is not cleared - https://github.com/alexmnyc/metrics/blob/16679f3e5fddcfc79bbe51bfab0ca8ab9c377088/metrics-graphite/src/main/java/com/codahale/metrics/graphite/Graphite.java#L114 no?